public/js/retraining.jsx

```diff
@@ -254,7 +254,7 @@ class WindowShade extends React.Component {
 class UpdateForm extends React.Component {
   constructor() {
     super();
-    this.state = { classCount: 0, files: {}, showFlash: false, checkboxes: new Set([]) };
```

Comment:
You replace the react state with a new thing, what about the rendering pipeline?